### DISCUSSION TITLE
Can Vike fetch data on server for first page load and handle subsequent fetching on client?

### DISCUSSION THREAD

**Discussion by sahidrahman404** [vikejs/vike#1160] [Discussion]
Hi, I want to use Vike with Relay. I'm wondering whether Vike can run GraphQL queries on the server side for the initial page load and then handle subsequent queries on the client side?

**Comment by brillout** [comment]
I'd use `onRenderHtml` (i.e. `render()` in `.page.server.js` if you aren't using the V1 design) which is always called on the server-side and only for the first page load. For subsequent client-side page navigation, I'd use `onBeforeRender()` (make sure it's isomorphic).

In case you didn't see it already: https://vike.dev/relay which includes an example of integrating Relay.

**Edit**: keep me udated how it goes. (Very much looking forward to a `vike-relay` integration package.)
